train.py

```diff
@@ -191,41 +191,48 @@ def train(hyp, opt, device, callbacks):  # hyp is path/to/hyp.yaml or hyp dictio
         model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model).to(device)
         LOGGER.info('Using SyncBatchNorm()')
 
-    # Trainloader
-    train_loader, dataset = create_dataloader(train_path,
-                                              imgsz,
-                                              batch_size // WORLD_SIZE,
-                                              gs,
-                                              single_cls,
-                                              hyp=hyp,
-                                              augment=True,
-                                              cache=None if opt.cache == 'val' else opt.cache,
-                                              rect=opt.rect,
-                                              rank=LOCAL_RANK,
-                                              workers=workers,
-                                              image_weights=opt.image_weights,
-                                              quad=opt.quad,
-                                              prefix=colorstr('train: '),
-                                              shuffle=True)
+    def _create_dataloaders():
+        # Trainloader
+        train_loader, dataset = create_dataloader(train_path,
```

Comment:
we don't want to call `create_dataloader` twice for the same dataset - this will [reinitialize the entire dataset](https://github.com/neuralmagic/yolov5/blob/edad59563782319739804c748aadad9e7b02ed1c/utils/datasets.py#L105) which is expensive. Instead, we should just be reinitializing the dataloader which is a wrapper over the dataset.

You should be able to pass in the dataset (can do dataloader.dataset if no access) and recreate the dataloader for a new batch size copying what happens here: https://github.com/neuralmagic/yolov5/blob/edad59563782319739804c748aadad9e7b02ed1c/utils/datasets.py#L116